### PR TITLE
[2.0] Allow installs along side symfony 3

### PR DESCRIPTION
**Pull request by GrahamCampbell**

### CODE REVIEW

**Comment by keradus** [comment]
then we need to run tests on both 2.x and 3.x lines at Travis

**Comment by GrahamCampbell** [comment]
Maybe, yeh.

**Comment by keradus** [comment]
Any news on that ?

**Comment by GrahamCampbell** [comment]
Give me a min.

**Comment by GrahamCampbell** [comment]
Ok, so let me explain what I've done in travis.

We now run tests on the following:
* 5.3 with lowest possible - that means symfony 2.1/2.3/2.5 is getting used
* 5.3 with stable - that means symfony 2.7, and later 2.8 is getting used
* 5.4 with stable - that means symfony 2.7, and later 2.8 is getting used
* 5.5 with unstable - that means symfony 3.x is getting used
* etc...

After symfony 3 is stable, the travis changes here can actually be reverted, due to the fact symfony 3 requires php 5.5, so our 5.3/5.4 builds will ALWAYS be using symfony 2 anyway.

File: .travis.yml [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
     fast_finish: true
     include:
         - php: 5.3
           env: 'COMPOSER_FLAGS="--prefer-stable --prefer-lowest"'
+        - php: 5.3
+          env: 'COMPOSER_FLAGS="--prefer-stable"'
+        - php: 5.4
+          env: 'COMPOSER_FLAGS="--prefer-stable"'
+        - php: 5.5
+          env: 'COMPOSER_FLAGS=""'
```
Comment (stof) [inline .travis.yml]: this can be omitted (bash is fine with env variables which don't exist, and will give you an empty string)

Comment (GrahamCampbell) [inline .travis.yml]: thanks

File: composer.json [hunk 2]
```diff
@@ -35,5 +35,6 @@
         "branch-alias": {
             "dev-master": "2.0-dev"
         }
-    }
+    },
+    "minimum-stability": "dev"
```
Comment (gharlan) [inline composer.json]: Wouldn't it be better to change the version constraints to `~2.3|~3.0@dev` etc. instead?

Comment (GrahamCampbell) [inline composer.json]: No, because those symfony dependencies have dependencies on others that mean that they'd end up loading 2.x when we wanted 3.x.

**Comment by GrahamCampbell** [comment]
Just rebased.

**Comment by keradus** [comment]
It's not intuitive for me when travis will use 2.x and when 3.x, isn't it possible to somehow decide about it explicitly ?

**Comment by GrahamCampbell** [comment]
Travis will always use 2.x on php 5.3/5.4 because 3.x requires php 5.5.9+.

**Comment by GrahamCampbell** [comment]
> isn't it possible to somehow decide about it explicitly ?

Yes and no. Yes, but not in a clean way. The best way imo would be to have 2 seperate composer.json files that we get travis to load for each case.

**Comment by GrahamCampbell** [comment]
I really don't like that idea here though. I think this way is the best. ;P

**Comment by keradus** [comment]
> Travis will always use 2.x on php 5.3/5.4 because 3.x requires php 5.5.9+.

And I really don't like implicitly behind that statement :(

**Comment by keradus** [comment]
What if we would like to test 2.x and 3.x on PHP 5.6 ?

**Comment by GrahamCampbell** [comment]
> What if we would like to test 2.x and 3.x on PHP 5.6 ?

Well, that's different. I don't think there's any point in doing that.

**Comment by keradus** [comment]
Why not? If we want a tool to run on both 2.x and 3.x then we may test it on 2.x and 3.x with same PHP  version.

**Comment by gharlan** [comment]
I don't like the `"minimum-stability": "dev"`. We should not use sf 2.x dev versions.

**Comment by GrahamCampbell** [comment]
We dont?

**Comment by GrahamCampbell** [comment]
@gharlan your suggestion would apply dev to the whole thing anyway, not just the 3.x bit

**Comment by stof** [comment]
IMO, we should have a single build running with ``"minimum-stability": "dev"``. See https://github.com/kriswallsmith/assetic/blob/56cb5d6dec9e7a68a4da2fa89844f39d41092f31/.travis.yml#L37 for the way to do it (this code assumes that there is no ``minimum-stability`` yet in the composer.json)

The project itself should not use it in its composer.json, otherwise released phars would also be built using dev versions, which is bad.

**Comment by GrahamCampbell** [comment]
I dont get what you mean. We dony want dev once 3.x is stable

**Comment by keradus** [comment]
> We dony want dev once 3.x is stable

Yeah, we don't want dev, and in this PR there is:
> "minimum-stability": "dev"

**Comment by GrahamCampbell** [comment]
Meh. I'll send a new PR once 3.x is actually stable.

**Comment by GrahamCampbell** [comment]
There's no massive rush for 3.x support as far as I can see.

**Comment by Soullivaneuh** [comment]
> Yeah, we don't want dev, and in this PR there is:
>
> > "minimum-stability": "dev"

Yes, but we could also add `"prefer-stable": true` or use `@dev` tag for travis.

**Pull request closed without merging**